tests/ApiTest.php

```diff
@@ -370,6 +370,52 @@ public function testUploadFromURL()
         }
     }
 
+    /**
+     * Test upload from URL
+     */
+    public function testUploadFromURLWithOptions()
+    {
+        $options = array(
+            'filename' => 'IMG_1.jpg',
+            'store' => true,
+            'check_status' => false,
+            'timeout' => 3,
+            'max_attempts' => 6,
+            'save_URL_duplicates' => true,
+            'check_URL_duplicates' => true,
+        );
+        try {
+            // We have to pre-upload in case file gets deleted from demo account
+            $file = $this->api->uploader->fromUrl('https://www.baysflowers.co.nz/wp-content/uploads/2015/06/IMG_9886_2.jpg', $options);
+        } catch (Exception $e) {
+            $this->fail('We get an unexpected exception trying to upload from url: '.$e->getMessage());
+        }
+        try {
+            $file = $this->api->uploader->fromUrl('https://www.baysflowers.co.nz/wp-content/uploads/2015/06/IMG_9886_2.jpg', $options);
+        } catch (Exception $e) {
+            $this->fail('We get an unexpected exception trying to upload from url: '.$e->getMessage());
+        }
+        $data = $file->__get('data');
+
+        $this->assertArrayHasKey('url', $data);
+        $this->assertArrayHasKey('uuid', $data);
```

Comment:
1. you can get a race condition here, that 1st upload is not yet finished, so you'll get an upload token, not data.
    I suggest to wait till upload is complete setting `check_status` to `true`.
2. we also want to check that the duplication check actually works and that `$uuid1 == $uuid2`